enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,325 @@
+  - TBD
+creation-date: 2022-12-06
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/MON-2483
+---
+
+# Scrape profiles
+
+## Summary
+
+The core OpenShift components ship a large number of metrics. A 4.12-nightly
+cluster on AWS currently produces around 350,000 unique timeseries by default,
+and enabling additional add-ons increases that number. Users have repeatedly
+asked for a supported method of making Prometheus consume less memory and CPU,
+either by increasing the scraping interval or by scraping fewer targets.
+
+These modifications are currently not possible, because the OpenShift operators
+manage their service/pod monitors (either directly or via the cluster version
+operator) and any manual modification would be reverted. This proposal outlines
+a solution for allowing users to set a level of scraping aligned with their
+needs.
+
+## Motivation
+
+OpenShift is an opinionated platform, and the default set of metrics has of
+course been crafted to match what we think the majority of users might need.
+Nevertheless, users have repeatedly asked for the ability to reduce the amount
+of memory consumed by Prometheus either by lowering the Prometheus scrape
+intervals or by modifying ServiceMonitors.
+
+Users currently can not control the ServiceMonitors scraped by Prometheus since some of the
+metrics collected are essential for other parts of the system to function propperly
+(console, HPA and alerts). Users also are not allowed to tune the interval at which Prometheus
+scrapes targets as this again can have unforeseen results that can hinder the platform, a very
+low cadence may overwhelm the platform Prometheus instance a very high interval may render some
+of the default alerts ineffective.
+
+The goal of this proposal is to allow users to pick their desired level of scraping while limiting
+the impact this might have on the platform, via resources under the control of the
+cluster-monitoring-operator and other platform operators.
+
+Furthermore to assess the viability of the scrape profile feature the monitoring
+team performed a detailed analysis of its impact in an OpenShift cluster.
+The analysis consisted in a test where an OpenShift cluster would run three replicas
+of the OpenShift Prometheus instance but each replica would be configured to a 
+different scrape profile (`full`, `minimal`, `uponly`). Then we would trigger a 
+workload using kube-burner and at the end of 2 hours, we evaluated the results.
+We concluded that in terms of resource usage, given the results obtained we
+can confidently state that the feature is quite valuable given the reduction of
+CPU by ~21% and memory by ~33% when comparing the `minimal` to the `full` profile.
+More detail can be consulted in this [document](https://docs.google.com/document/d/1MA-HTJQ_X7y_bwpJS2IPbGmC4qDyIMp25jisr34X2F4/edit?usp=sharing)
+
```

Comment:
I've found additional data for this section based on some Telemetry numbers: since a few releases, we collect for each cluster the top-3 Prometheus jobs generating most samples.
Based on the 4.11 numbers, the 5 components most often reported are: the Kubernetes API servers, the Kubernetes schedulers, kube-state-metrics, kubelet and the networ daemon.

Reply:
That's neat! Can you share the query that gave you that info?